### PR TITLE
Week 4 Comments

### PR DESCRIPTION
**Pull request by nicollecastrog**
Great job Janine, you came so close to doing it the way the solution does and you ended up going beyond what I ever intended you guys to do as part of this assignment! 😄

### CODE REVIEW

File: Week04/lesson08/homework/temp_converter/index.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+<head>
+  <meta charset="UTF-8">
+  <title>Temperature Converter - Part 1</title>
+</head>
+
+<body>
+  <h1>Temperature Converter</h1>
+
+  <div id="inputs">
+    <input id="celciusInput" value="20" />
```
Comment (nicollecastrog) [inline Week04/lesson08/homework/temp_converter/index.html:12]: Default value for the input, always nice option to give the user that 👍

File: Week04/lesson08/homework/temp_converter/js/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+function convert() {
+
+
+  var c = document.querySelector("#celciusInput").value;
+  var f = c * 1.8 + 32
```
Comment (nicollecastrog) [inline Week04/lesson08/homework/temp_converter/js/index.js:5]: Don't forget the semi-colon at the end of the line!

File: Week04/lesson08/homework/temp_converter/js/index.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  c = parseInt(c);
+  f = parseInt(f);
+  // I included this, but seems to work without it also. Confused!
```
Comment (nicollecastrog) [inline Week04/lesson08/homework/temp_converter/js/index.js:9]: Right, to answer your question:

So, in the same way that you can do:

`4 + ""`

...and get: `"4"` as a string, Javascript is nice sometimes about "figuring out" what you mean. So while sometimes it'll blow up for you trying to do maths with a string (text), other times, like here, it'll let it slide.

Basically, when `c * 1.8 + 32` occurs, JS is being nice and converting `c` into a number for you.

That said, it's nice to be thorough as well as guard against an evil user that might try to put text instead of numbers into that `<input>`, so doing `parseFloat` on `c` is a good idea. I say `parseFloat` and not `parseInt` (which is what you've done) because you need to consider the possibility that the user might put a decimal value in.

Also, you only need to convert `c`, because `c` is the only value you're pulling off the DOM. `f`'s value is determined by `c` and the maths done on `c`, so you don't need to `parse` it in any way.

So basically, you're looking at:

```
var c = document.querySelector("#celciusInput").value;
var f = parseFloat(c) * 1.8 + 32;
```

File: Week04/lesson08/homework/temp_converter_2/js/index.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+function convertCToF() {
```
Comment (nicollecastrog) [inline Week04/lesson08/homework/temp_converter_2/js/index.js:1]: So I think your biggest issue with this other version is a conceptional one more than an execution one. Basically, I think you got all tangled up by the fact that there's only ONE "convert" button but TWO possible things that could happen (convert C to F or convert F to C).

Back to basics: like in example one, you have ONE button that's going to trigger ONE function as a result:

```
function convert() {
  //Do stuff here
} 
document.querySelector("#submit").onclick = convert;
```

... whether you're translating C to F or vice-versa is logic that goes WITHIN that single `convert()` function.

I'll cover this in class tomorrow, but consider this:

```
function convert() {
  var c = document.querySelector('#c').value;
  var f = document.querySelector('#f').value;
  var result, scale;

  if (f === "") {
    //If the value of f is an empty string, then we know the user put the text into the c <input> and is looking to convert INTO f
  } else {
    //Since the value of f is NOT an empty string, then we know the value of c must be, so then we know the user put the text into the f <input> and is looking to convert INTO c
  }
  document.querySelector('h3').innerHTML = "Result: " + result + "°" + scale;
}
```

Comment (nicollecastrog) [inline Week04/lesson08/homework/temp_converter_2/js/index.js:1]: That said, very cool that you got it converting upon typing in the input, good googling! 👍
